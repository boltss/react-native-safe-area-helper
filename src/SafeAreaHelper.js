// @flow
import React, { PureComponent } from 'react'
import {
	SafeAreaView,
	View,
	StyleSheet,
	type LayoutChangeEvent,
	Dimensions,
	Platform,
} from 'react-native'

import { getInsetFromSafeAreaLayout } from './util/SafeAreaUtil'


export type SafeAreaInsets = {
	top: number,
	bottom: number,
	left: number,
	right: number,
}
type Props = {
	onInsetsChange: (insets: SafeAreaInsets) => void,
}

const styles = StyleSheet.create({
	safeArea: {
		...StyleSheet.absoluteFillObject,
	},
	insideSafeArea: {
		flex: 1,
	},
})

export default class SafeAreaHelper extends PureComponent<Props> {
	onInsideSafeAreaLayout = (event: LayoutChangeEvent) => {
		const { onInsetsChange } = this.props

		const insets = getInsetFromSafeAreaLayout(event.nativeEvent.layout)
		onInsetsChange(insets)
	}

	props: Props

	render() {
		return (
			<SafeAreaView style={styles.safeArea} pointerEvents='none'>
				<View
					style={styles.insideSafeArea}
					onLayout={this.onInsideSafeAreaLayout}
				/>
			</SafeAreaView>
		)
	}
}
