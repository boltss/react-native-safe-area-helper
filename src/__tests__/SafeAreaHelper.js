// @flow
import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

import SafeAreaHelper from '../SafeAreaHelper'


describe('SafeAreaHelper', () => {
	it('renders correctly', () => {
		const tree = renderer.create(
			<SafeAreaHelper />,
		).toJSON()
		expect(tree).toMatchSnapshot()
	})
})
