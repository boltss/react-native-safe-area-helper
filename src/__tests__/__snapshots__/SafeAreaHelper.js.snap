// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SafeAreaHelper renders correctly 1`] = `
<RCTSafeAreaView
  pointerEvents="none"
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
</RCTSafeAreaView>
`;
